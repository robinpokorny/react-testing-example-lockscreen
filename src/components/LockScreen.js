import React, { PropTypes } from "react";
import ClockDisplay from "./ClockDisplay";
import TopOverlay from "./TopOverlay";
import SlideToUnlock from "./SlideToUnlock";

export default class LockScreen extends React.Component {
  static propTypes = {
    wallpaperPath: PropTypes.string,
    userInfoMessage: PropTypes.string,
    onUnlocked: PropTypes.func,
  };

  render() {
    const {
      wallpaperPath,
      userInfoMessage,
      onUnlocked,
    } = this.props;

    return (
      <div
        style={{
          height: "100%",
          display: "flex",
          justifyContent: "space-between",
          flexDirection: "column",
          backgroundImage: wallpaperPath ? `url(${wallpaperPath})` : "",
          backgroundColor: "black",
          backgroundPosition: "center",
          backgroundSize: "cover",
        }}
      >
        <ClockDisplay now={new Date(...[2017, 2, 23])}/>
        {userInfoMessage ? (
          <TopOverlay
            style={{
              padding: "2em",
              marginBottom: "auto",
            }}
          >
            {userInfoMessage}
          </TopOverlay>
        ) : null}
        <SlideToUnlock onSlide={onUnlocked} />
      </div>
    );
  }
}
