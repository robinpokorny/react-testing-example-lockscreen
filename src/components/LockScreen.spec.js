// Uses Jest Snapshots
import React from "react";
import renderer from 'react-test-renderer'

import LockScreen from "./LockScreen";

describe("LockScreen", () => {

  test('renders without props', () => {
    const tree = renderer.create(
      <LockScreen />
    ).toJSON()

    expect(tree).toMatchSnapshot('no-props')
  })

  test('renders with all props passed', () => {
    const tree = renderer.create(
      <LockScreen
        onUnlocked={jest.fn()}
        userInfoMessage='This is my favorite phone!'
        wallpaperPath='some/image.png'
      />
    ).toJSON()

    expect(tree).toMatchSnapshot('with-props')
  })

});
