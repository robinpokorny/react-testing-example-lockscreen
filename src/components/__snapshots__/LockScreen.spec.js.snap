exports[`no-props 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "backgroundImage": "",
      "backgroundPosition": "center",
      "backgroundSize": "cover",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "space-between",
    }
  }>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(55, 55, 55, 0.5)",
        "color": "white",
        "display": "flex",
        "fontSize": "48px",
        "height": "2em",
        "justifyContent": "center",
        "width": "100%",
      }
    }>
    00:00:00
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(55, 55, 55, 0.5)",
        "color": "white",
        "display": "flex",
        "fontSize": 16,
        "height": 64,
        "justifyContent": "center",
        "userSelect": "none",
        "width": "100%",
      }
    }>
    <div
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.3)",
          "borderRadius": 8,
          "height": 32,
          "lineHeight": "32px",
          "maxWidth": "400px",
          "position": "relative",
          "textAlign": "center",
          "width": "calc(100% - 32px)",
        }
      }>
      <div
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(169, 169, 169)",
            "borderRadius": 8,
            "color": "rgb(84, 84, 84)",
            "cursor": "pointer",
            "fontSize": 24,
            "height": 32,
            "left": 0,
            "lineHeight": "32px",
            "position": "absolute",
            "textAlign": "center",
            "transition": "left 0.5s ease-out",
            "width": 48,
          }
        }>
        >
      </div>
      Slide to unlock
    </div>
  </div>
</div>
`;

exports[`with-props 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "backgroundImage": "url(some/image.png)",
      "backgroundPosition": "center",
      "backgroundSize": "cover",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "space-between",
    }
  }>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(55, 55, 55, 0.5)",
        "color": "white",
        "display": "flex",
        "fontSize": "48px",
        "height": "2em",
        "justifyContent": "center",
        "width": "100%",
      }
    }>
    00:00:00
  </div>
  <div
    style={
      Object {
        "background": "linear-gradient(rgba(85, 85, 85, 0.5), transparent)",
        "color": "white",
        "marginBottom": "auto",
        "padding": "2em",
      }
    }>
    This is my favorite phone!
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(55, 55, 55, 0.5)",
        "color": "white",
        "display": "flex",
        "fontSize": 16,
        "height": 64,
        "justifyContent": "center",
        "userSelect": "none",
        "width": "100%",
      }
    }>
    <div
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.3)",
          "borderRadius": 8,
          "height": 32,
          "lineHeight": "32px",
          "maxWidth": "400px",
          "position": "relative",
          "textAlign": "center",
          "width": "calc(100% - 32px)",
        }
      }>
      <div
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(169, 169, 169)",
            "borderRadius": 8,
            "color": "rgb(84, 84, 84)",
            "cursor": "pointer",
            "fontSize": 24,
            "height": 32,
            "left": 0,
            "lineHeight": "32px",
            "position": "absolute",
            "textAlign": "center",
            "transition": "left 0.5s ease-out",
            "width": 48,
          }
        }>
        >
      </div>
      Slide to unlock
    </div>
  </div>
</div>
`;
